apps/bfd-pipeline/bfd-pipeline-rif-load/src/main/java/gov/cms/bfd/pipeline/rif/load/RifLoader.java

```diff
@@ -596,6 +614,313 @@ private void processAsync(
+      }
+
+      beneficiaryMonthly =
+          getBeneficiaryMonthly(
+              beneficiaryRecord,
+              LocalDate.of(year, 11, 1),
+              beneficiaryRecord.getEntitlementBuyInNovInd(),
+              beneficiaryRecord.getFipsStateCntyNovCode(),
+              beneficiaryRecord.getHmoIndicatorNovInd(),
+              beneficiaryRecord.getMedicaidDualEligibilityNovCode(),
+              beneficiaryRecord.getMedicareStatusNovCode(),
+              beneficiaryRecord.getPartCContractNumberNovId(),
+              beneficiaryRecord.getPartCPbpNumberNovId(),
+              beneficiaryRecord.getPartCPlanTypeNovCode(),
+              beneficiaryRecord.getPartDContractNumberNovId(),
+              beneficiaryRecord.getPartDLowIncomeCostShareGroupNovCode(),
+              beneficiaryRecord.getPartDPbpNumberNovId(),
+              beneficiaryRecord.getPartDRetireeDrugSubsidyNovInd(),
+              beneficiaryRecord.getPartDSegmentNumberNovId());
+
+      if (beneficiaryMonthly != null) {
+        currentYearBeneficiaryMonthly.add(beneficiaryMonthly);
+      }
+
+      beneficiaryMonthly =
+          getBeneficiaryMonthly(
+              beneficiaryRecord,
+              LocalDate.of(year, 12, 1),
+              beneficiaryRecord.getEntitlementBuyInDecInd(),
+              beneficiaryRecord.getFipsStateCntyDecCode(),
+              beneficiaryRecord.getHmoIndicatorDecInd(),
+              beneficiaryRecord.getMedicaidDualEligibilityDecCode(),
+              beneficiaryRecord.getMedicareStatusDecCode(),
+              beneficiaryRecord.getPartCContractNumberDecId(),
+              beneficiaryRecord.getPartCPbpNumberDecId(),
+              beneficiaryRecord.getPartCPlanTypeDecCode(),
+              beneficiaryRecord.getPartDContractNumberDecId(),
+              beneficiaryRecord.getPartDLowIncomeCostShareGroupDecCode(),
+              beneficiaryRecord.getPartDPbpNumberDecId(),
+              beneficiaryRecord.getPartDRetireeDrugSubsidyDecInd(),
+              beneficiaryRecord.getPartDSegmentNumberDecId());
+
+      if (beneficiaryMonthly != null) {
+        currentYearBeneficiaryMonthly.add(beneficiaryMonthly);
+      }
+
+      if (currentYearBeneficiaryMonthly.size() > 0) {
+        List<BeneficiaryMonthly> currentBeneficiaryMonthlyWithUpdates;
+
+        Beneficiary beneficiaryFromDb =
+            entityManager.find(Beneficiary.class, beneficiaryRecord.getBeneficiaryId());
+
+        if (beneficiaryFromDb != null && beneficiaryFromDb.getBeneficiaryMonthlys().size() > 0) {
+          currentBeneficiaryMonthlyWithUpdates = beneficiaryFromDb.getBeneficiaryMonthlys();
+          List<BeneficiaryMonthly> currentYearBeneficiaryMonthlyPrevious =
+              beneficiaryFromDb.getBeneficiaryMonthlys().stream()
+                  .filter(e -> year == e.getYearMonth().getYear())
+                  .collect(Collectors.toList());
+
+          if (currentYearBeneficiaryMonthlyPrevious.size() > 0) {
```

Comment:
FWIW, you don't need this `if` -- the for loop just won't run if the list is empty.

Reply:
Changed